src/metabase/db/data_migrations.clj

```diff
@@ -172,6 +176,67 @@
                                  [:like
                                   :dashcard.visualization_settings "%\"click_link_template\":%"]]})))
 
+(defn- raw-setting
+  "Get raw setting directly from DB.
+  For some reasons during data-migration [[metabase.models.setting/get]] return the default value defined in
+  [[metabase.models.setting/defsetting]] instead of value from Setting table."
+  [k]
+  (db/select-one-field :value Setting :key (name k)))
+
+(defn- remove-group-id-from-mappings-by-setting-key
+  [mapping-setting-key]
+  (let [admin-group-id (:id (group/admin))
+        mapping        (try
+                        (json/parse-string (raw-setting mapping-setting-key))
+                        (catch Exception _e
+                          {}))]
+    (when-not (empty? mapping)
+      (db/update! Setting (name mapping-setting-key)
+                  :value
+                  (->> mapping
+                          (map (fn [[k v]] [k (filter #(not= admin-group-id %) v)]))
+                          (into {})
+                          (json/generate-string))))))
+
+(defn- auth-method-configured?
+  "Replicate the checks from
+  [[metabase.integrations.ldap/ldap-configured?]]
+  [[metabase-enterprise.sso.integrations.sso-settings/jwt-configured]]
+  [[metabase-enterprise.sso.integrations.sso-settings/saml-configured]]
+  The difference is that:
+  - We get the settings directly from DB
+  - We could perform this check for jwt,saml even in OSS version"
+  [method]
+  (case method
+    :ldap (boolean (and (= (raw-setting :ldap-enabled) "true")
+                        (raw-setting :ldap-host)
+                        (raw-setting :ldap-user-base)))
+
+    :jwt  (boolean (and (= (raw-setting :jwt-enabled) "true")
+                        (raw-setting :jwt-identity-provider-uri)
+                        (raw-setting :jwt-shared-secret)))
+
+    :saml (boolean (and (= (raw-setting :saml-enabled) "true")
+                        (raw-setting :saml-identity-provider-uri)
+                        (raw-setting :saml-identity-provider-certificate)))))
```

Comment:
Not a huge fan of duplicating all this logic here. Why do we need to check this stuff at all anyway? Can't we just check whether there is a Setting named `jwt-group-mappings` and update it if it exists?